internals/src/macros.rs

```diff
@@ -205,7 +205,10 @@ macro_rules! impl_from_infallible {
     }
 }
 
-/// Implements `to_hex` for functions that have implemented [`core::fmt::LowerHex`]
+/// Adds an implementation of `pub fn to_hex(&self) -> String` if `alloc` feature is enabled.
+///
+/// The added function allocates a `String` then calls through to [`core::fmt::LowerHex`].
+/// Calling this macro without the `alloc` feature enabled is a noop.
 #[macro_export]
 #[cfg(feature = "alloc")]
 macro_rules! impl_to_hex_from_lower_hex {
```

Comment:
I've just realized the macro is broken.

If crate `a` depends on `internals` and has `alloc` off and then crate `b` depends on `internals` turning on the `alloc` feature crate `a` has suddenly injected code referencing a non-existing `alloc` crate and thus `a` breaks.

There's another small but ugly issue that the `alloc` requirement is not properly documented in the calling crate.

Because of the second issue, out of two possible fixes - using `$crate::alloc` and just putting the `cfg` on the `to_hex` method and declaring the assumption of the macro that the calling crate has an `alloc` feature that also activates the `alloc` crate, I prefer the latter.

This is obviously unrelated to the PR.